test/e2e/common.go

```diff
@@ -0,0 +1,370 @@
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// the klusterlets should be cleaned up
+	Eventually(func() error {
+		klusterlets, err := operatorClient.OperatorV1().Klusterlets().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(klusterlets.Items) != 0 {
+			return fmt.Errorf("the klusterlets are not deleted")
+		}
+		return nil
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+
+	// clean the managed clusters
+	clusters, err := clusterClient.ClusterV1().ManagedClusters().List(context.TODO(), metav1.ListOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	for _, cluster := range clusters.Items {
+		err = clusterClient.ClusterV1().ManagedClusters().Delete(context.TODO(), cluster.Name, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// the managed clusters should be cleaned up
+	Eventually(func() error {
+		clusters, err := clusterClient.ClusterV1().ManagedClusters().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(clusters.Items) != 0 {
+			return fmt.Errorf("the manged clusters are not deleted")
+		}
+		return nil
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+
+	// the pods in open-cluster-management-agent ns should be cleaned up
+	Eventually(func() error {
+		var err error
+		pods, err := kubeClient.CoreV1().Pods(klusterletDefaultNamespace).
+			List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(pods.Items) != 0 {
+			return fmt.Errorf("the pods in ns %v are not deleted", klusterletDefaultNamespace)
+		}
+		return nil
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+}
+
+func checkHubReady() {
+	// make sure open-cluster-management-hub namespace is created
+	Eventually(func() error {
+		_, err := kubeClient.CoreV1().Namespaces().
+			Get(context.TODO(), clusterManagerNamespace, metav1.GetOptions{})
+		return err
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+
+	// make sure hub deployments are created
+	Eventually(func() error {
+		_, err := kubeClient.AppsV1().Deployments(clusterManagerNamespace).
+			Get(context.TODO(), hubRegistrationDeployment, metav1.GetOptions{})
+		return err
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+
+	Eventually(func() error {
+		_, err := kubeClient.AppsV1().Deployments(clusterManagerNamespace).
+			Get(context.TODO(), hubWebhookDeployment, metav1.GetOptions{})
+		return err
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+}
+
+func checkKlusterletOperatorReady() {
+	// make sure klusterlet operator deployment is created
+	Eventually(func() error {
+		_, err := kubeClient.AppsV1().Deployments(operatorNamespace).
+			Get(context.TODO(), klusterletOperator, metav1.GetOptions{})
+		return err
+	}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+
+	// make sure open-cluster-management-agent namespace is created
```

Comment:
why this specific namespace?

Reply:
delete this specific namespace, and user can customize the klusterlet agent namespace